packages/std/src/traits.rs

```diff
@@ -71,20 +73,27 @@ pub trait Api: Copy + Clone + Send {
 pub type QuerierResult = SystemResult<ApiResult<Binary>>;
 
 pub trait Querier: Clone + Send {
-    // Note: ApiError type can be serialized, and the below can be reconstituted over a WASM/FFI call.
-    // Since this is information that is returned from outside, we define it this way.
-    //
-    // ApiResult is a format that can capture this info in a serialized form. We parse it into
-    // a typical Result for the implementing object
-    fn query(&self, request: &QueryRequest) -> QuerierResult;
+    // TODO: rename this... right now this is demo to see if we can pipe it through unparsed in the VM
+    fn raw_query(&self, bin_request: &[u8]) -> QuerierResult;
+
+    fn query<T: Serialize>(&self, request: &QueryRequest<T>) -> QuerierResult {
+        match to_vec(request) {
+            Ok(raw) => self.raw_query(&raw),
+            // TODO: maybe I want to make this a SystemError::InvalidRequest ?
+            Err(e) => Ok(Err(e.into())),
+        }
+    }
 
     /// Makes the query and parses the response.
     /// Any error (System Error, Error or called contract, or Parse Error) are flattened into
     /// one level. Only use this if you don't have checks on other side.
     ///
     /// eg. When querying another contract, you will often want some way to detect/handle if there
     /// is no contract there.
-    fn parse_query<T: DeserializeOwned>(&self, request: &QueryRequest) -> StdResult<T> {
+    fn parse_query<T: Serialize, U: DeserializeOwned>(
+        &self,
+        request: &QueryRequest<T>,
+    ) -> StdResult<U> {
```

Comment:
I think it would be cool to inline the content of `query` here and rename `parse_query` -> `query`.

Then we just need two functions: 1. all typed, 2. all raw